server/src/main/kotlin/org/jetbrains/teamcity/vault/server/VaultBuildStartContextProcessor.kt

```diff
@@ -31,39 +31,45 @@ class VaultBuildStartContextProcessor(private val connector: VaultConnector) : B
     companion object {
         val LOG = Logger.getInstance(Loggers.SERVER_CATEGORY + "." + VaultBuildStartContextProcessor::class.java.name)!!
 
-        private fun getFeature(build: SBuild): VaultFeatureSettings? {
-            val buildType = build.buildType ?: return null
+        private fun getFeature(build: SBuild): List<VaultFeatureSettings> {
+            val buildType = build.buildType ?: return emptyList()
 
-            val connectionFeature = buildType.project.getAvailableFeaturesOfType(OAuthConstants.FEATURE_TYPE).firstOrNull {
+            val connectionFeatures = buildType.project.getAvailableFeaturesOfType(OAuthConstants.FEATURE_TYPE).filter {
```

Comment:
Let's carefully process all returned features from `getAvailableFeaturesOfType` method. 
1. There's possibility that feature with same prefix could exist in project and it's parent project. Some test is required for that.
2. Now you could have two project features with same prefix. It should be processed somehow and probably some error shown.

Reply:
You can also just make two features on the same project, I haven't implemented any uniqueness-check. We can check if there are multiple features with the same prefix and log an error in case there are.

Reply:
I've resolved that and pushed commit into your master branch